app/controllers/swaps_controller.rb

```diff
@@ -10,11 +10,11 @@ def index
   end
 
   def my_offers
-    @swaps = Offer.where(:user => selected_user, :status => ['pending','cancelled','closed', 'accepted', 'withdrawn', 'not accepted', 'declined', 'unsuccessful', 'deleted']).joins(:roster).order("rosters.start desc")
+    @swaps = Offer.where(:user => selected_user, :status => ['pending','cancelled','closed', 'accepted', 'withdrawn', 'not accepted', 'declined', 'unsuccessful', 'deleted', 'rejected']).where("offers.roster_id is NULL").joins(:request).order('offers.roster_id DESC NULLS FIRST').sort_by{|offer| offer.request.roster.start} + Offer.where(:user => selected_user, :status => ['pending','cancelled','closed', 'accepted', 'withdrawn', 'not accepted', 'declined', 'unsuccessful', 'deleted', 'rejected']).where("offers.roster_id is not NULL").joins(:request).order('offers.roster_id DESC NULLS FIRST').sort_by{|offer| offer.request.roster.start}
   end
 
   def my_requests
-    @swaps = Request.joins(:roster).where(:user => selected_user, :status => 'open').order("rosters.start")
```

Comment:
I think this is wrong. If I am a Swap Requestor, I must have a Roster. Removing this will cause the requests to be out of order which is not correct.